assets/js/securesubmit.js

```diff
@@ -1,103 +1,378 @@
 /*jslint browser:true, unparam:true*/
 /*global hps, gformInitSpinner*/
 (function (window, $) {
-  'use strict';
-
-  window.SecureSubmit = function (args) {
-    var prop;
-
-    this.form = null;
-    this.apiKey = null;
-    this.formId = null;
-    this.ccFieldId = null;
-    this.ccPage = null;
-    this.isAjax = null;
-
-    for (prop in args) {
-      if (args.hasOwnProperty(prop)) {
-        this[prop] = args[prop];
-      }
-    }
-
-    this.init = function () {
-      var SecureSubmitObj = this;
-
-      if (!this.isCreditCardOnPage()) {
-        return;
-      }
-
-      // initialize spinner
-      if (!this.isAjax) {
-        gformInitSpinner(this.formId);
-      }
-
-      // bind SecureSubmit functionality to submit event
-      $('#gform_' + this.formId).submit(function (event) {
-        var $form = $(this);
-        var ccInputPrefix = 'input_' + SecureSubmitObj.formId + '_' + SecureSubmitObj.ccFieldId + '_';
-
-        SecureSubmitObj.form = $form;
-
-        if ($('#securesubmit_response').size() === 0) {
-
-          var hps = new Heartland.HPS({
-            publicKey: SecureSubmitObj.apiKey,
-            cardNumber: $form.find('#' + ccInputPrefix + '1').val().replace(/\D/g, ''),
-            cardCvv: $form.find('#' + ccInputPrefix + '3').val(),
-            cardExpMonth: $form.find('#' + ccInputPrefix + '2_month').val(),
-            cardExpYear: $form.find('#' + ccInputPrefix + '2_year').val(),
-            success: function (response) {
-              SecureSubmitObj.secureSubmitResponseHandler(response);
-            },
-            error: function (response) {
-              SecureSubmitObj.secureSubmitResponseHandler(response);
-            }
-          });
+    'use strict';
+
+    window.SecureSubmit = function (args) {
+        var prop;
 
-          hps.tokenize();
+        this.form = null;
+        this.apiKey = null;
+        this.formId = null;
+        this.ccFieldId = null;
+        this.ccPage = null;
+        this.isAjax = null;
 
-          return false;
+        for (prop in args) {
+            if (args.hasOwnProperty(prop)) {
+                this[prop] = args[prop];
+            }
         }
 
-        return true;
-      });
+        this.init = function () {
+
+            var SecureSubmitObj = this;
+            Heartland.Card.attachNumberEvents('#cc_number');
+            Heartland.Card.attachExpirationEvents('#_exp_date');
+            Heartland.Card.attachCvvEvents('#_cvv_number');
```

Comment:
@slogsdon For the sake of sanity i am restoring the version of this from master till we are ready with the new CC stuff